src/librustdoc/passes/doc_test_lints.rs

```diff
@@ -67,13 +67,28 @@ crate fn should_have_doc_example(cx: &DocContext<'_>, item: &clean::Item) -> boo
                 | clean::ImportItem(_)
                 | clean::PrimitiveItem(_)
                 | clean::KeywordItem(_)
+                // check for trait impl
+                | clean::ImplItem(clean::Impl { trait_: Some(_), .. })
         )
     {
         return false;
     }
+
     // The `expect_def_id()` should be okay because `local_def_id_to_hir_id`
     // would presumably panic if a fake `DefIndex` were passed.
     let hir_id = cx.tcx.hir().local_def_id_to_hir_id(item.def_id.expect_def_id().expect_local());
+
+    // check if parent is trait impl
+    if let Some((_, parent_node)) = cx.tcx.hir().parent_iter(hir_id).next() {
+        use rustc_hir::*;
```

Comment:
I'm not a big fan of imports in the middle of the code, can you instead import at the top (of the function?) `use rustc_hir as hir` and then prepend the `hir` types with it please?